neem-interface.pl

```diff
@@ -2,70 +2,148 @@
+mem_event_set_failed(Action) :- kb_project(action_failed(Action)).
 
-mem_event_begin(Event) :- get_time(CurrentTime),tell(occurs(Event) since CurrentTime),!.
+mem_event_set_succeeded(Action) :- kb_project(action_succeeded(Action)).
 
-%belief_perceived_at(ObjectType, Frame, Object) :- get_time(CurrentTime),execution_agent(Agent),tell([has_type(Object,ObjectType),is_at(Object,Frame) since CurrentTime]).
+mem_event_add_diagnosis(Situation, Diagnosis) :- kb_project(satisfies(Situation, Diagnosis)).
 
-belief_perceived_at(ObjectType, Mesh, Rotation, Object) :- tell([has_type(Object,ObjectType),has_type(ShapeRegion, soma:'MeshShape'), has_type(Shape, soma:'Shape'), triple(Object, soma:'hasShape', Shape), triple(Shape, dul:'hasRegion', ShapeRegion), triple(ShapeRegion, soma:'hasFilePath', Mesh),has_type(Origin,soma:'Origin'),triple(ShapeRegion,'http://knowrob.org/kb/urdf.owl#hasOrigin',Origin),triple(Origin, 'http://www.ease-crc.org/ont/SOMA.owl#hasPositionVector', term([0.0,0.0,0.0])),triple(Origin, 'http://www.ease-crc.org/ont/SOMA.owl#hasOrientationVector',term(Rotation))]).
-
-belief_perceived_at(ObjectType, Object) :- tell([has_type(Object,ObjectType)]).
-
-
-add_participant_with_role(Action, ObjectId, RoleType) :- 
-ask([executes_task(Action, Task),triple(Event,dul:'hasTimeInterval',TimeInterval),triple(TimeInterval,soma:'hasIntervalBegin',Start),triple(TimeInterval,soma:'hasIntervalEnd',End)]), 
-tell([has_participant(Action,ObjectId), has_type(Role, RoleType), has_role(ObjectId,Role) during Action,task_role(Task, Role)]).
-
-%add_participant_with_role(Action, ObjectId, RoleType) :- ask(executes_task(Action, Task)), tell([has_participant(Action,ObjectId), has_type(Role, RoleType), has_role(ObjectId,Role) during [0.0,0.0]]).
-
-add_parameter(Task,ParameterType,RegionType) :- tell([has_type(Parameter, ParameterType), has_type(Region,RegionType),has_assignment(Parameter,Region) during [0.0,0.1], has_parameter(Task, Parameter)]).
-
-add_grasping_parameter(Action,GraspingOrientationType) :- ask(executes_task(Action, Task)), tell([has_type(GraspingOrientation,GraspingOrientationType), has_type(GraspingOrientationConcept,'http://www.ease-crc.org/ont/SOMA.owl#GraspingOrientation'), has_parameter(Task,GraspingOrientationConcept),holds(GraspingOrientationConcept, dul:classifies, GraspingOrientation),has_region(Action,GraspingOrientation)]),!.
-
-add_comment(Entity,Comment) :- tell(triple(Entity, 'http://www.w3.org/2000/01/rdf-schema#comment', Comment)).
+mem_add_subaction_with_task(ParentAction,SubActionType,TaskType,SubAction) :-
+    execution_agent(Agent),
+    kb_project([
+        new_iri(SubAction, SubActionType), is_individual(SubAction), instance_of(SubAction,SubActionType),
```

Comment:
Hey, @NeumannM89 is looking into using the new code and we had problems with this line. What value does SubActionType have in your executions? Generally a subaction should always be a soma:'Action' or at most one of it's subclasses.

Reply:
In my executions, SubActionType is 'http://www.ease-crc.org/ont/SOMA.owl#PhysicalAction'. Task type is something like 'http://www.ease-crc.org/ont/SOMA.owl#Grasping', and ParentAction is the IRI of the top-level action individual of the NEEM. SubAction is left uninitialized, because new_iri(SubAction, SubActionType) creates a new IRI for it.

Reply:
Ok, I was concerned, because I thought there was some confusion with Action and Task, but this is the correct usage.